modules/storage-api/src/main/java/org/apache/ignite/internal/storage/StorageDestroyedException.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.storage;
+
+import org.apache.ignite.lang.ErrorGroups.Storage;
+
+/**
+ * Exception that is be thrown when trying to access a storage that is being destroyed or is already destroyed.
+ */
+public class StorageDestroyedException extends StorageException {
+    private static final long serialVersionUID = -7988332521347221109L;
+
+    /**
+     * Default constructor.
+     */
+    public StorageDestroyedException() {
+        this("Storage is already destroyed");
+    }
+
+    /**
+     * Constructor.
+     *
+     * @param message Error message.
+     */
+    public StorageDestroyedException(String message) {
+        super(Storage.ALREADY_CLOSED_ERR, message);
```

Comment:
Maybe we need a new code for the error, otherwise it's a little strange?

Reply:
To be honest, I'm not sure. Does the user need to know the difference between a destroyed and closed storage? From the user's perspective, it's a storage with which no interactions are possible as it's 'not available anymore'. The existing code seems to be fine for both cases.

Reply:
It’s at your discretion, of course, but for me it looks like: I want to go home, but I can’t because the house is closed or destroyed, no, it’s closed in any case.